src/main/java/com/github/kumaraman21/intellijbehave/service/JBehaveUtil.java

```diff
@@ -106,113 +73,80 @@ public String compute() {
+                    .ifPresent(annotationTexts::add);
 
-    @NotNull
-    private static Set<String> getStepAnnotationTexts(@NotNull PsiAnnotation stepAnnotation) {
-        final String annotationText = AnnotationUtil.getStringAttributeValue(stepAnnotation, "value");
-
-        if (annotationText == null) {
-            return ImmutableSet.of();
+            findAnnotation(annotations, Aliases.class)
+                    .map(JBehaveUtil::getAliasesAnnotationTexts)
+                    .ifPresent(annotationTexts::addAll);
         }
 
-        return new PatternVariantBuilder(annotationText).allVariants();
+        return annotationTexts.stream()
+                .map(PatternVariantBuilder::new)
+                .map(PatternVariantBuilder::allVariants)
+                .flatMap(Set::stream)
+                .collect(Collectors.toSet());
     }
 
-    @NotNull
-    private static Set<String> getAliasAnnotationTexts(@NotNull PsiAnnotation aliasAnnotation) {
-        final String annotationText = AnnotationUtil.getStringAttributeValue(aliasAnnotation, "value");
-
-        if (annotationText == null) {
-            return ImmutableSet.of();
-        }
-
-        return new PatternVariantBuilder(annotationText).allVariants();
+    private static Optional<String> getAnnotationText(@NotNull PsiAnnotation annotation) {
+        return Optional.ofNullable(AnnotationUtil.getStringAttributeValue(annotation, "value"));
     }
 
     @NotNull
     private static Set<String> getAliasesAnnotationTexts(@NotNull PsiAnnotation aliasAnnotation) {
         PsiAnnotationMemberValue values = aliasAnnotation.findAttributeValue("values");
 
         if (!(values instanceof PsiArrayInitializerMemberValue)) {
-            return ImmutableSet.of();
+            return Collections.emptySet();
         }
 
         final PsiArrayInitializerMemberValue attrValue = (PsiArrayInitializerMemberValue) values;
 
-        if (attrValue == null) {
-            return ImmutableSet.of();
-        }
-
         final PsiConstantEvaluationHelper constantEvaluationHelper = JavaPsiFacade.getInstance(aliasAnnotation.getProject()).getConstantEvaluationHelper();
 
-        return from(asList(attrValue.getInitializers()))
-                .transform(new Function<PsiAnnotationMemberValue, String>() {
-                    @Override
-                    public String apply(@Nullable PsiAnnotationMemberValue psiAnnotationMemberValue) {
-                        Object constValue = constantEvaluationHelper.computeConstantExpression(psiAnnotationMemberValue);
-                        return constValue instanceof String ? (String) constValue : null;
-                    }
-                }).transformAndConcat(TO_A_SET_OF_PATTERNS).toSet();
+        return Stream.of(attrValue.getInitializers())
+                .map(constantEvaluationHelper::computeConstantExpression)
+                .filter(Objects::nonNull)
+                .map(String.class::cast)
```

Comment:
not exactly like how it was
before: only cast to string if its instanceof, otherwise return null
now: only cast to string if its not null, regardless of actual type (prone to ClassCastException)